server/data/data-jmap/src/main/java/org/apache/james/jmap/mime4j/SizeUtils.java

```diff
@@ -0,0 +1,99 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.mime4j;
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+import org.apache.james.mime4j.dom.Body;
+import org.apache.james.mime4j.dom.Entity;
+import org.apache.james.mime4j.dom.SingleBody;
+import org.apache.james.mime4j.message.BodyPart;
+import org.apache.james.mime4j.message.DefaultMessageWriter;
+import org.apache.james.mime4j.message.MessageImpl;
+
+import com.google.common.io.CountingOutputStream;
+
+public class SizeUtils {
+    public static long sizeOf(Entity entity) throws IOException {
```

Comment:
The `Email/get` test suite was imcomplete. I got on a production workload a call for `MultipartImp` with some FakeBinaryBody in it.

Regarding the JMAP spec, `partId` and `blobId` are null for multiparts so there is no downloads.

To quote the spec:

```
(as referenced by the blobId, i.e., the number of octets in the file the user would download).
```

So I fixed this by returning `0` for multiparts as they cannot be downloaded...

Reply:
Is this handled?

Reply:
When I read your comment, I thought you did, as it finished with a sentence `So I fixed this...`

Did I misunderstand something?

Reply:
Well then you need to be more clear in your comments then. 

Will take a look, with your suggested tests for other multiparts that seem to test this